src/internal_events/common.rs

```diff
@@ -90,3 +90,36 @@ impl InternalEvent for AwsBytesSent {
         );
     }
 }
+
+#[derive(Debug)]
+pub struct StreamClosedError {
+    pub error: String,
+    pub count: usize,
+}
+
+impl InternalEvent for StreamClosedError {
```

Comment:
Just to note: I left some comments/questions about `StreamClosedError` in the[ EventsStoreDB Metrics PR](https://github.com/vectordotdev/vector/pull/10890)